terraform/modules/cloudwatch/variable.tf

```diff
@@ -0,0 +1,30 @@
+variable "resource_prefix" {
+  description = "Prefix for all resources"
+   type        = string
+}
+
+variable "ecs_cluster_name" {
```

Comment:
these should have default null values and can be validated in locals like:

locals {

  validate_ecs_cluster = (var.target_type == "ecs" && var.ecs_cluster_name == null) ? tobool("Please pass in the ecs cluster name if you are using ecs as the target.") : true

}

this way there will be an error if the target is set to ecs and these vars don't have values passed in